### PR TITLE
Add ability to summarize selected intents using Preprocessor

### PR DESCRIPTION
**Pull request by adithyabsk**
## Related Issue
Fixes issue #14 

### Description
* Add column_summary field to BaseIntent and fix broken tests
* Implement column_summary field for NumericIntent and CategoricalIntent
* Implement summarize functionality for Preprocessor using selected intents and column_summaries
* Fix more broken tests, it's tough life keeping 100% coverage :)

### CODE REVIEW

File: foreshadow/intents/general.py [hunk 1]
```diff
@@ -11,6 +13,25 @@
...
+    """Computes the mode and the most frequent values
+
+        Args:
+            series (pandas.Series): the series to analyze
+            count (int): the n number of most frequent values
+
+    """
+
+    vc = series.value_counts()
+    if series[~series.isnull()].nunique() == 1:
```
Comment (jichaogp) [inline foreshadow/intents/general.py]: Why do we need this special path? The rest of the code should still work even if there is only one unique value, no?

File: foreshadow/intents/general.py [hunk 2]
```diff
@@ -11,6 +13,25 @@
...
+        return None, []
+    else:
+        mode = series.mode().values.tolist()
+        if len(mode) == 1:
+            mode = mode[0]
+        return (mode, vc.nlargest(count).reset_index().values.tolist())
```
Comment (jichaogp) [inline foreshadow/intents/general.py]: Can we calculate the percentage frequency as well?

File: foreshadow/intents/general.py [hunk 3]
```diff
@@ -66,6 +92,55 @@ def is_intent(cls, df):
...
+
+        """
+        data = df.ix[:, 0]
+        nan_num = int(data.isnull().sum())
+        invalid_num = int(
+            pd.to_numeric(df.ix[:, 0], errors="coerce").isnull().sum() - nan_num
+        )
+        outliers = (
+            data[np.abs(data - data.mean()) > (3 * data.std())]
+            .nlargest(10)
```
Comment (jichaogp) [inline foreshadow/intents/general.py]: this will only take the positive side, right?

File: foreshadow/tests/test_intents/test_general.py [hunk 4]
```diff
@@ -21,6 +30,88 @@ def test_numeric_intent_is_intent():
...
+    import pandas as pd
+    from foreshadow.intents import mode_freq
+
+    np.random.seed(0)
+    X0 = pd.Series([0])
+    X_ = pd.Series([0, 10, 10])
+    X1 = pd.Series([0, 10, 10, 20, 20])
+    X2 = pd.Series(np.random.randint(0, 10, 100))
+
+    assert mode_freq(X0) == (None, [])
```
Comment (jichaogp) [inline foreshadow/tests/test_intents/test_general.py]: Shall we expect it return (0, [0, 1])?

File: foreshadow/intents/general.py (hunk 3 above)
Comment (adithyabsk) [reply foreshadow/intents/general.py]: that's a great catch

File: foreshadow/tests/test_intents/test_general.py (hunk 4 above)
Comment (adithyabsk) [reply foreshadow/tests/test_intents/test_general.py]: Fixed

**Pull request merged**